### PR TITLE
choose a host in the same pool that the service runs in to query for …

### PR DESCRIPTION
**Pull request by stug**
…smartstack backends

### CODE REVIEW

File: tests/test_smartstack_tools.py [hunk 1]
```diff
@@ -255,8 +255,8 @@ def test_get_replication_for_instance(
     system_paasta_config,
 ):
     mock_mesos_slaves = [
-        {'hostname': 'host1', 'attributes': {'region': 'fake_region1'}},
-        {'hostname': 'host2', 'attributes': {'region': 'fake_region1'}},
+        {'hostname': 'host1', 'attributes': {'region': 'fake_region1', 'pool': 'default'}},
+        {'hostname': 'host2', 'attributes': {'region': 'fake_region1', 'pool': 'default'}},
```
Comment (mattmb) [inline tests/test_smartstack_tools.py]: Can you expand the test a bit to test the new logic?

I think we could assert that `get_replication_for_all_services` is called with the correct host?

Comment (mattmb) [reply tests/test_smartstack_tools.py]: But ship anyway if it is a real pain to do so for some reason :-)

**Comment by mattmb** [comment]
Would you also mind amending the commit message to say "why"

File: tests/test_smartstack_tools.py (hunk 1 above)
Comment (stug) [reply tests/test_smartstack_tools.py]: Yup good call, went too fast.

**Comment by EvanKrall** [comment]
Oops, was looking at an out of date diff, you already added the test. Ship it!

**Pull request merged**